openquakeplatform/openquakeplatform/static/js/irv_viewer.js

```diff
@@ -14,1043 +14,864 @@
-    var selLayer = document.getElementById('layer-list');
-
-    // Create a header for the menu drop down
-    var catMenuHeader = document.createElement('option');
-    catMenuHeader.innerHTML = 'Projects:';
-    selCat.appendChild(catMenuHeader);
-
-    $.getJSON(TILESTREAM_API_URL,
-    function(json) {
-
-        var name, cat, type, grids, pDef;
-
-        // Create the category list (build the object)
-        for (var i=0; i < json.length; i++) {
-            name = json[i].mapped_value;
-            cat = json[i].category;
-            type = json[i].type;
-            grids = json[i].grids;
-            pDef = json[i].project_definition;
-
-            if (cat != undefined && type == 'svir-qgis') {
-                categoryList.push(cat);
-                layerNames[name] = [];
-                layersByCat[cat] = [];
-                projectDefinition[name] = [];
-
-                if (grids != undefined) {
-                    var grid = grids.toString();
-                    var gridName = grid.split('/')[4];
-                    layerGrids.push(gridName);
+    // Match each primary indicator with it's respective data value
+    var primaryRiskIndicatorObj = {};
+    for (var ic = 0; ic < la.length; ic++) {
+        var tempObj = la[ic].properties;
+        for (var d = 0; d < indicatorChildrenKey.length; d++) {
+            for(var o in tempObj) {
+                if (indicatorChildrenKey[d] == o) {
+                    var tempValue = indicatorChildrenKey[d];
+                    var tempValue2 = tempObj[o];
+                    var tempValue3 = tempObj[selectedRegion];
+                    if (primaryRiskIndicatorObj[tempValue] == undefined) {
+                        primaryRiskIndicatorObj[tempValue] = tempValue2;
+                        primaryRiskIndicatorObj.district = tempValue3;
+                    } else {
+                        primaryRiskIndicatorObj[tempValue] = primaryRiskIndicatorObj[tempValue] + "," +tempValue2;
+                        primaryRiskIndicatorObj.district = primaryRiskIndicatorObj.district + "," +tempValue3;
+                    }
                 }
             }
-            if (grids != undefined) {
-                layerNames[grids] = [];
-            }
         }
-
-        // Create the category list (population the object)
-        for (var j=0; j < json.length; j++) {
-            name = json[j].mapped_value;
-            cat = json[j].category;
-            type = json[j].type;
-            grids = json[j].grids;
-            pDef = json[j].project_definition;
-
-            if (cat != undefined && type == 'svir-qgis') {
-                layerId = json[j].id;
-                layerTitle = json[j].mapped_value;
-                layerNames[name].push(layerId);
-                layersByCat[cat].push(layerTitle);
-                projectDefinition[name].push(pDef);
+    }
+    return primaryRiskIndicatorObj;
+}
+
+function createIndex(la, index) {
```

Comment:
Please add a docstring also here, specifying the meaning of the parameters, and the return value. About the return value, it looks like you are building an indicator object and returning it, so you might refactor the function into createIndicator